spec/unit/inventory_v4_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_relative '../spec_helper'
+
+module Hawkular::Inventory::RSpecUnit
+  include Hawkular::InventoryV4
+  context 'Inventory v4 unit test' do
+
+    let(:resource_hash) do
+      {
+        "id" => "root",
+        "type" => {},
+        "children" => [
+          { "id" => "child-01", "type" => { "id" => "type-01"} },
+          { "id" => "child-02", "type" => { "id" => "type-02"},
+            "children"=> [
+              { "id" => "child-03", "type" => { "id" => "type-02"} },
+              { "id" => "child-04", "type" => { "id" => "type-02"} },
+              { "id" => "child-05", "type" => { "id" => "type-03"} }
+            ]
+          },
+          { "id" => "child-06", "type" => { "id" => "type-01"} }
+        ]
+      }
+    end
+
+    describe 'children by type' do
+      it 'should return direct children' do
+        r = Resource.new(resource_hash)
+        expect(r.children_by_type('type-02').size).to eql(1)
+      end
+
+      it 'should return 0 when direct children type is not found' do
+        r = Resource.new(resource_hash)
+        expect(r.children_by_type('type-03').size).to eql(0)
+      end
+
+      it 'should work recursive' do
+        r = Resource.new(resource_hash)
+        expect(r.children_by_type('type-02', recursive=true).size).to eql(3)
```

Comment:
Named arguments in Ruby do not work like this. Also, by convention we use `eq` instead of `eql`, because `eql` means equality by identity, not equality by comparison.